test/test_other.py

```diff
@@ -3884,6 +3884,30 @@ def test_js_lib_quoted_key(self):
 
     self.do_runf(test_file('hello_world.c'), 'hello, world!', emcc_args=['--js-library', 'lib.js'])
 
+  def test_js_lib_proxying(self):
+    # Regression test for a bug we had where jsifier would find and use
+    # the inner function in a library function consisting of a single
+    # line arrow function.
+    # See https://github.com/emscripten-core/emscripten/issues/20264
+    create_file('lib.js', r'''
+addToLibrary({
+  foo__proxy: 'sync',
+  foo: () => setTimeout(() => {
+    console.log('should not see this before "done"');
+  }),
```

Comment:
Isn't this a little racey? A timeout of 0 means it will run in the next event loop iteration, but the runtime might exit before that? That is, we might get only "main" in the output, sometimes.

Reply:
Oh, wait, I misread this, `done` is not from here but from below.

Reply:
I think the problem is that the main thread which calls `foo` could hit its even loop at print "should not see this before "done"" from the setTimeout before the secondary thread sees that `foo` is complete and allow itself to continue executing and print "done"

Reply:
The key here is that "foo" is proxied to the main thread.

Reply:
Ah, yes... it must be the proxying, yeah, good point.